src/Components/Sighting.js

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import React, { useEffect, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import Card from 'react-bootstrap/Card';
+
+import { BACKEND_URL } from '../constants.js';
+
+const Sighting = () => {
+  const [sightingIndex, setSightingIndex] = useState();
+  const [sighting, setSighting] = useState();
+
+  useEffect(() => {
+    // If there is a sightingIndex, retrieve the sighting data
+    if (sightingIndex) {
+      axios
+        .get(`${BACKEND_URL}/sightings/${sightingIndex}`)
+        .then((response) => {
+          setSighting(response.data);
+        });
+    }
+    // Only run this effect on change to sightingIndex
+  }, [sightingIndex]);
+
+  // Update sighting index in state if needed to trigger data retrieval
+  const params = useParams(); // gets the number from the url
+  if (sightingIndex !== params.sightingIndex) {
+    setSightingIndex(params.sightingIndex);
+  }
```

Comment:
Why not just trust the param sightingIndex? Why the need to check for different value? Seems a bit redundant to me but maybe there is some reason